### PR TITLE
Fix DHCP commands

### PR DESCRIPTION
**Pull request by pederhan**

### CODE REVIEW

File: mreg_cli/commands/dhcp.py [hunk 1]
```diff
@@ -110,15 +105,24 @@ def disassoc(args: argparse.Namespace) -> None:
...
+                ) from None
 
             ipaddress_to_use = ips_with_mac[0]
+    else:
+        ipobjs = IPAddress.get_by_ip(addr.address)
+        if not ipobjs:
+            raise InputFailure(f"IP address {args.name} does not exist.")
+        elif len(ipobjs) > 1:
+            raise InputFailure(f"IP {args.name} is in use by {len(ipobjs)} hosts.")
+        ipaddress_to_use = ipobjs[0]
```
Comment (pederhan) [inline mreg_cli/commands/dhcp.py:122]: This is not great! Nested try/excepts is rarely a good sign

Comment (pederhan) [inline mreg_cli/commands/dhcp.py:122]: This is not great! Nested try/excepts is rarely a good sign

File: mreg_cli/commands/dhcp.py [hunk 2]
```diff
@@ -110,15 +105,24 @@ def disassoc(args: argparse.Namespace) -> None:
...
 
-    mac = ipaddress_to_use.macaddress
     ipaddress_to_use.disassociate_mac()
+    mac = ipaddress_to_use.macaddress
```
Comment (pederhan) [inline mreg_cli/commands/dhcp.py:125]: This is not great! Nested try/excepts is rarely a good sign

**Comment by terjekv** [comment]
`get_host_by_any_means` allows for fetching by IP, but the problem here is that we need the exact IP for use later. We could do a fetch via `get_host_by_any_means` and then cycle through the IPs if we were given an IP (or a Mac actually, same issue arises then).

File: mreg_cli/commands/dhcp.py [hunk 3]
```diff
@@ -85,41 +101,34 @@ def disassoc(args: argparse.Namespace) -> None:
...
+                raise InputFailure(
+                    f"Host {host} has multiple IP addresses with MAC addresses."
+                ) from None
 
-            ipaddress_to_use = ips_with_mac[0]
+            ipaddress = ips_with_mac[0]
 
-    mac = ipaddress_to_use.macaddress
-    ipaddress_to_use.disassociate_mac()
+    ipaddress.disassociate_mac()
```
Comment (terjekv) [inline mreg_cli/commands/dhcp.py:129]: Nitpick. I always thought it looked weird to first something like `disassociate_mac()` and then fetch it via `.macaddress`.

File: mreg_cli/api/fields.py [hunk 4]
```diff
@@ -47,6 +47,11 @@ class IPAddressField(FrozenModel):
 
     address: IP_AddressT
 
+    @classmethod
+    def from_string(cls, address: str) -> IPAddressField:
```
Comment (terjekv) [inline mreg_cli/api/fields.py:51]: We should migrate to this everywhere. Very good idea!

**Review by terjekv — APPROVED** [review]

File: mreg_cli/commands/dhcp.py [hunk 5]
```diff
@@ -23,6 +24,25 @@ def __init__(self, cli: Any) -> None:
         super().__init__(cli, command_registry, "dhcp", "Manage DHCP associations.", "Manage DHCP")
 
 
+def ipaddress_from_ip_arg(arg: str) -> IPAddress | None:
```
Comment (terjekv) [inline mreg_cli/commands/dhcp.py:27]: Add a line about what can be raised? Also use EntityOwnershipMismatch when multiple hosts are found to use the IP? Something like:

```
:raises: InputFailure when the IP does not exist.
:raises: EntityOwnershipMismatch when the IP is in use by multiple hosts.
```

File: mreg_cli/commands/dhcp.py (hunk 3 above)
Comment (pederhan) [reply mreg_cli/commands/dhcp.py:132]: I've removed this assignment altogether. We can access it directly when constructing the message.

**Pull request merged**